server/index.mjs

```diff
@@ -341,37 +355,63 @@ const loop = () => {
       }
     }
 
-    if (player.selectedPlayer) {
+    const { selectedPlayer } = player;
+
+    if (selectedPlayer) {
       if (player.settings.follow) {
         player.updateFollowing(map, players);
       }
 
       if (player.settings.fight && player.canAttack({ PF, finder, map })) {
         player.attackDelayTicks = 0;
         player.energyUse("attack");
-        player.attack = player.selectedPlayer.name;
+        player.attack = selectedPlayer.name;
 
-        const hit = getCurrentWeapon(player.equipment.weapon).weapon.attack;
+        const currentWeapon = getCurrentWeapon(player.equipment.weapon);
+        const hit = currentWeapon.weapon.attack;
+        const skillDetails = currentWeapon.skill;
 
-        player.selectedPlayer.gotHit(hit);
+        player.skillUpdate(skillIncrease(player.skills, skillDetails));
 
-        if (player.hasRangedWeapon() && player.useArrow()) {
-          io.to(player.socketId).emit(
-            "items:update",
-            player.backpack,
-            player.equipment
+        io.to(player.socketId).emit(
+          "skills:update",
+          shapeSkillsForClient(player.skills)
+        );
+
+        if (selectedPlayer.equipment.shield) {
+          const selectedPlayerSkillDetails = getCurrentWeapon(
+            selectedPlayer.equipment.shield
+          ).skill;
+
+          selectedPlayer.skillUpdate(
+            skillIncrease(selectedPlayer.skills, selectedPlayerSkillDetails)
+          );
+
+          io.to(selectedPlayer.socketId).emit(
```

Comment:
This was already emitted in line 376, remove first emit and move this emit after condition, just leave skill increase in here